python/ray/tune/experiment.py

```diff
@@ -105,6 +106,16 @@ def from_json(cls, name, spec):
         if "run" not in spec:
             raise TuneError("No trainable specified!")
 
+        if "repeat" in spec:
+            warnings.warn("repeat is deprecated, use total_samples instead")
+            spec["total_samples"] = spec["repeat"]
+            del spec["repeat"]
+        
+        if "grid_search" in spec or "config" in spec and "grid_search" in spec["config"]:
```

Comment:
Why remove this? This is a very useful behavior.

Reply:
So first and foremost "repeat" needs to be changed; it's been too often the source of user errors. Hence "total_samples", as per discussion on ray-dev.

Then, if you specify "total_samples" to be `N` as the number of total evaluations, and your grid search totals to `D` different evaluations, then I don't think it makes sense to have `N*D` evaluations - it should be either `N` or `D`. 

If you want to _really_ want to repeat a grid search, you can do `tune.grid_search(range(iterations))` or we can consider adding a parameter for grid search specifically.

Reply:
How about renaming `total_samples` => `resample` or `num_samples` to avoid the "total" connotation?

It is very common to want to repeat a grid search N times. Wouldn't it be weird for this parameter to silently do nothing in that case?

Reply:
`num_samples` is probably fine, and we keep the grid search semantics.